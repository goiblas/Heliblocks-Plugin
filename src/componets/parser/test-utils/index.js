import React from 'react';
import { render } from '@testing-library/react';
import { Store } from "../store";
import { SlotFillProvider } from '@wordpress/components';
import { SlotFillProvider as CustomSlotFillProvider } from '../../slotfill';

const AllTheProviders = ({ children }) => {
  return (
    <SlotFillProvider>
      <CustomSlotFillProvider id="1">
        {children}
      </CustomSlotFillProvider>
    </SlotFillProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }

export const store = new Store();