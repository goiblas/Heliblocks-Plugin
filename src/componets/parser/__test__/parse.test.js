import Parser, { getCurrentVersion, getProcessors } from "../index";
import { render, store } from '../test-utils';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

describe('Parser lib', () => {
    let processors,
        version;
    beforeEach(() => {
        const versions = getCurrentVersion();
        processors = getProcessors(versions.processors);
        version = versions.parser;
    });
    afterEach(() => {
        store.clear();
    });
    test('should reemplace text line', () => {
        const html = "<div>hello world!</div>";
      
        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        const save = render(<Parser version={version} html={html} processors={processors} environment="save" store={store} />)
        expect(edit.container.firstChild).toMatchSnapshot();
        expect(save.container.firstChild).toMatchSnapshot();
    });

    test('should reemplace image', () => {
        const html = "<div><div>hello world!</div><img src='image.jpg' alt='alt' /></div>";
      
        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        const save = render(<Parser version={version} html={html} processors={processors} environment="save" store={store} />)
        expect(edit.container.firstChild).toMatchSnapshot();
        expect(save.container.firstChild).toMatchSnapshot();
    });
    test('should reemplace multiline tag', () => {
        const html = "<div><ul><li>hello</li><li>world</li></div>";
      
        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        const save = render(<Parser version={version} html={html} processors={processors} environment="save" store={store} />)
        expect(edit.container.firstChild).toMatchSnapshot();
        expect(save.container.firstChild).toMatchSnapshot();
    });
    test('should replace attributes to props', () => {
        const html = "<div class='classname'>hello world!</div>";
      
        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        const save = render(<Parser version={version} html={html} processors={processors} environment="save" store={store} />)
        expect(edit.container.firstChild).toMatchSnapshot();
        expect(save.container.firstChild).toMatchSnapshot();
    });

    test('should not parse svg ', () => {
        const html = '<svg xmlns="http://www.w3.org/2000/svg" width="500" height="500"><g><path d="M12.9" /></g></svg>';
        
        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        const save = render(<Parser version={version} html={html} processors={processors} environment="save" store={store} />)
        
        expect(edit.container.firstChild).toMatchSnapshot();
        expect(save.container.firstChild).toMatchSnapshot();
    });

    test('should replace label', () => {
        const html = '<div><label>Label</label></div>';

        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        const save = render(<Parser version={version} html={html} processors={processors} environment="save" store={store} />)
        
        expect(edit.container.firstChild).toMatchSnapshot();
        expect(save.container.firstChild).toMatchSnapshot();
    });
    test('should replace complex estructures', () => {
        const html = '<div class="wrapper"><div class="image"><img src="https://source.unsplash.com/random" width="500"></div><div class="content"><h2>hola mundo</h2><p>lorem</p><p>lorem</p></div></div>';

        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)
        
        expect(edit.container.firstChild).toMatchSnapshot();
    });
    
    test('should replace orphan links', () => {
        const html = '<div><p>hello</p><a href="#">Orphan link</a></div>';

        const edit = render(<Parser version={version} html={html} processors={processors} environment="edit" store={store} />)

        expect(edit.container.firstChild).toMatchSnapshot();
    });
});