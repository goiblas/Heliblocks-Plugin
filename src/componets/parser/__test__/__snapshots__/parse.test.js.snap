// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser lib should not parse svg  1`] = `
<svg
  height="500"
  width="500"
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <path
      d="M12.9"
    />
  </g>
</svg>
`;

exports[`Parser lib should not parse svg  2`] = `
<svg
  height="500"
  width="500"
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <path
      d="M12.9"
    />
  </g>
</svg>
`;

exports[`Parser lib should reemplace image 1`] = `
<div>
  <div
    aria-multiline="true"
    class="rich-text block-editor-rich-text__editable"
    contenteditable="true"
    role="textbox"
    style="white-space: pre-wrap;"
  >
    hello world!
  </div>
  <img
    alt="alt"
    src="image.jpg"
  />
</div>
`;

exports[`Parser lib should reemplace image 2`] = `
<div>
  <div>
    <div>
      hello world!
    </div>
  </div>
  <img
    alt="alt"
    src="image.jpg"
  />
</div>
`;

exports[`Parser lib should reemplace multiline tag 1`] = `
<div>
  <ul>
    <li
      aria-multiline="true"
      class="rich-text block-editor-rich-text__editable"
      contenteditable="true"
      role="textbox"
      style="white-space: pre-wrap;"
    >
      hello
    </li>
    <li
      aria-multiline="true"
      class="rich-text block-editor-rich-text__editable"
      contenteditable="true"
      role="textbox"
      style="white-space: pre-wrap;"
    >
      world
    </li>
  </ul>
</div>
`;

exports[`Parser lib should reemplace multiline tag 2`] = `
<div>
  <ul>
    <li>
      <div>
        hello
      </div>
    </li>
    <li>
      <div>
        world
      </div>
    </li>
  </ul>
</div>
`;

exports[`Parser lib should reemplace text line 1`] = `
<div
  aria-multiline="true"
  class="rich-text block-editor-rich-text__editable"
  contenteditable="true"
  role="textbox"
  style="white-space: pre-wrap;"
>
  hello world!
</div>
`;

exports[`Parser lib should reemplace text line 2`] = `
<div>
  <div>
    hello world!
  </div>
</div>
`;

exports[`Parser lib should replace attributes to props 1`] = `
<div
  aria-multiline="true"
  class="rich-text block-editor-rich-text__editable classname"
  contenteditable="true"
  role="textbox"
  style="white-space: pre-wrap;"
>
  hello world!
</div>
`;

exports[`Parser lib should replace attributes to props 2`] = `
<div
  class="classname"
>
  <div>
    hello world!
  </div>
</div>
`;

exports[`Parser lib should replace complex estructures 1`] = `
<div
  class="wrapper"
>
  <div
    class="image"
  >
    <img
      src="https://source.unsplash.com/random"
      width="500"
    />
  </div>
  <div
    class="content"
  >
    <h2
      aria-multiline="true"
      class="rich-text block-editor-rich-text__editable"
      contenteditable="true"
      role="textbox"
      style="white-space: pre-wrap;"
    >
      hola mundo
    </h2>
    <p
      aria-multiline="true"
      class="rich-text block-editor-rich-text__editable"
      contenteditable="true"
      role="textbox"
      style="white-space: pre-wrap;"
    >
      lorem
    </p>
    <p
      aria-multiline="true"
      class="rich-text block-editor-rich-text__editable"
      contenteditable="true"
      role="textbox"
      style="white-space: pre-wrap;"
    >
      lorem
    </p>
  </div>
</div>
`;

exports[`Parser lib should replace label 1`] = `
<div
  aria-multiline="true"
  class="rich-text block-editor-rich-text__editable"
  contenteditable="true"
  role="textbox"
  style="white-space: pre-wrap;"
>
  <label>
    Label
  </label>
  
</div>
`;

exports[`Parser lib should replace label 2`] = `
<div>
  <div>
    <label>
      Label
    </label>
  </div>
</div>
`;

exports[`Parser lib should replace orphan links 1`] = `
<div>
  <p
    aria-multiline="true"
    class="rich-text block-editor-rich-text__editable"
    contenteditable="true"
    role="textbox"
    style="white-space: pre-wrap;"
  >
    hello
  </p>
  <div
    aria-multiline="true"
    class="rich-text block-editor-rich-text__editable"
    contenteditable="true"
    role="textbox"
    style="white-space: pre-wrap;"
  >
    <a
      class=""
      href="#"
    >
      Orphan link
    </a>
    
  </div>
</div>
`;
